lib/Solver/Z3Builder.cpp

```diff
@@ -400,16 +401,20 @@ Z3ASTHandle Z3Builder::getInitialArray(const Array *root) {
     array_expr = buildArray(unique_name.c_str(), root->getDomain(),
                             root->getRange());
 
-    if (root->isConstantArray()) {
-      // FIXME: Flush the concrete values into Z3. Ideally we would do this
-      // using assertions, which might be faster, but we need to fix the caching
-      // to work correctly in that case.
-      for (unsigned i = 0, e = root->size; i != e; ++i) {
-        Z3ASTHandle prev = array_expr;
-        array_expr = writeExpr(
-            prev, construct(ConstantExpr::alloc(i, root->getDomain()), 0),
-            construct(root->constantValues[i], 0));
+    if (root->isConstantArray()
+        && constant_array_assertions.count(root) == 0) {
+     std::vector<Z3ASTHandle> array_assertions;
+     for (unsigned i = 0, e = root->size; i != e; ++i) {
+        //construct(= (select i root) root->value[i]) to be asserted in Z3Solver.cpp
+        int width_out;
+        Z3ASTHandle array_value = construct(root->constantValues[i],&width_out);
```

Comment:
Please run clang-format on this entire patch. I can see that this hasn't been done.

Reply:
Done